src/interceptors/request.ts

```diff
@@ -109,4 +113,53 @@ export class CacheRequestInterceptor<D>
 
     return config;
   };
+
+  private isMethodAllowed = (
+    axios: AxiosCacheInstance,
+    method: Method = 'get',
+    properties?: Partial<CacheProperties>
+  ): boolean => {
+    const requestMethod = method.toLowerCase();
+    const allowedMethods = properties?.methods || axios.defaults.cache.methods;
+
+    for (const method of allowedMethods) {
+      if (method.toLowerCase() === requestMethod) {
+        return true;
+      }
+    }
+
+    return false;
+  };
+
+  private setRequestHeaders = (
+    cache: StaleStorageValue | EmptyStorageValue,
+    config: CacheRequestConfig<D> & { cache: Partial<CacheProperties> | undefined }
+  ) => {
+    if (!config.cache) return;
+    config.headers ||= {};
+
+    const { etag, modifiedSince } = config.cache;
+
+    if (etag) {
+      const etagValue = etag === true ? cache.data?.headers[Header.ETag] : etag;
+      if (etagValue) {
+        config.headers[Header.IfNoneMatch] = etagValue;
+      }
+    }
+
+    if (
+      modifiedSince &&
+      cache.state === 'stale'
+
+      // TODO: See if this only applies to get and head
+      // && (!config.method ||
+      // config.method.toLowerCase() === 'get' ||
+      // config.method.toLowerCase() === 'head')
+    ) {
+      const modifiedDate =
+        modifiedSince === true ? new Date(cache.createdAt) : modifiedSince;
+
+      config.headers[Header.IfModifiedSince] = modifiedDate.toUTCString();
+    }
+  };
```

Comment:
You also need to modify axios validateStatus to accept 304 (default is only 200-299 range)

Reply:
As you can see here, there's no need to run testCachePredicate again. As this value is only stale because it passed the test the first time.

https://github.com/ArthurFiorette/axios-cache-interceptor/blob/3b879475f19c276eba258b1546e7676eb416af1d/src/interceptors/response.ts#L49

Reply:
I'm not sure I see the relation with my message. You're expecting a 304 status from target server, however by default axios throws an error when status code it not between 200 and 299, so you need to modify `config` to accept status 304 (if either `if-modified-since` or `if-none-match` headers are set 

```
config.validateStatus = function (status) {
            return status==304 || !previousValidateStatus || previousValidateStatus(status);
          }
```

Reply:
Oh, you mean really axios, not this library. yeah i totally forgot about that

Reply:
Fixed? https://github.com/ArthurFiorette/axios-cache-interceptor/blob/5e868108cf986aee05ea38cf08aed3e69532ebc7/src/interceptors/request.ts#L75-L77

Reply:
maybe i have added an infinite loop...

https://github.com/ArthurFiorette/axios-cache-interceptor/blob/48887e4b7a2ae9a59d4517de4d32b07e6cd48afe/src/interceptors/request.ts#L75-L78

Reply:
don't add `(status >= 200 && status < 300)`, users can (not sure why but hey...) disable some status by configuring `config.validateStatus` (or even set it to false), 

`return oldValidate?.(status) || status === 304;` is enough and clearly convey what we're looking for :)

Reply:
My intention was, when there is no validateStatus defined, axios not to report an error if the response comes with a status other than 304. Let me fix it